# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/debug/Frame-type-01.js
# Commit: d70ec1e79887
# Full Hash: d70ec1e7988710de0b79351f730a2498972cd7e0
# Author: Logan Smyth <loganfsmyth@gmail.com>
# Date: 2019-12-09 09:50:39
# Regressor Bug: 1600204
# File Overlap Count: 1
# Description:
#   Bug 1600204 - Part 1: Support .type on suspended generator frames. r=jimb
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D55231
# ==============================================================================

diff -r e3354d109039 -r d70ec1e79887 js/src/jit-test/tests/debug/Frame-type-01.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/debug/Frame-type-01.js	Fri Dec 06 02:29:18 2019 +0000
@@ -0,0 +1,22 @@
+// Debugger.Frame.prototype.type on a normal function.
+
+load(libdir + "asserts.js");
+
+const g = newGlobal({ newCompartment: true });
+const dbg = Debugger(g);
+
+g.eval(`
+function f() {}
+`);
+
+let frame;
+dbg.onEnterFrame = function(f) {
+  frame = f;
+  assertEq(frame.type, "call");
+};
+
+g.f();
+
+assertEq(!!frame, true);
+// Throws because the frame has terminated.
+assertThrowsInstanceOf(() => frame.type, Error);